tests/runtime/http_server/test_http_server.py

```diff
@@ -265,14 +335,73 @@ def test_docs_with_models(
     assert response.status_code == 200
 
 
+def test_uvicorn_server_config_valid():
+    """Make sure that arbitrary uvicorn configs can be passed through from
+    runtime.http.server_config
+    """
+    timeout_keep_alive = 10
+    with temp_config(
+        {
+            "runtime": {
+                "http": {"server_config": {"timeout_keep_alive": timeout_keep_alive}}
+            }
+        },
+        "merge",
+    ):
+        server = http_server.RuntimeHTTPServer()
+        assert server.server.config.timeout_keep_alive == timeout_keep_alive
+
+
+def test_uvicorn_server_config_invalid_tls_overlap():
+    """Make sure uvicorn TLS arguments cannot be set if TLS is enabled in caikit
+    config
+    """
+    with temp_config(
+        {
+            "runtime": {
+                "http": {
+                    "server_config": {
+                        "ssl_keyfile": "/some/file.pem",
+                    }
+                }
+            }
+        },
+        "merge",
+    ):
+        with generate_tls_configs(port=1234, tls=True, mtls=True):
+            with pytest.raises(ValueError):
+                http_server.RuntimeHTTPServer()
+
+
+def test_uvicorn_server_config_invalid_kwarg_overlap():
+    """Make sure uvicorn config can't be set for configs that caikit manages"""
+    with temp_config(
+        {
+            "runtime": {
+                "http": {
+                    "server_config": {
+                        "log_level": "debug",
+                    }
+                }
+            }
+        },
+        "merge",
+    ):
+        with pytest.raises(ValueError):
+            http_server.RuntimeHTTPServer()
+
+
+## Inference Tests #############################################################
+
+
 def test_inference_sample_task(sample_task_model_id, client):
     """Simple check that we can ping a model"""
     json_input = {"inputs": {"name": "world"}, "model_id": sample_task_model_id}
     response = client.post(
         f"/api/v1/task/sample",
         json=json_input,
     )
-    json_response = json.loads(response.content.decode(response.default_encoding))
```

Comment:
This was a rampant anti-pattern in this file, so I fixed it everywhere to avoid folks wondering down the line why some used `response.json()` and others used this complex parsing